pkg/reconcile/pipeline/handler/collect/impl_test.go

```diff
@@ -745,6 +746,105 @@ var _ = Describe("Collect From Provisioned Service", func() {
 
 })
 
+var _ = Describe("Collect From Direct Secret", func() {
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(GinkgoT())
+		ctx = mocks.NewMockContext(mockCtrl)
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	Describe("successful processing", func() {
+		var (
+			services []pipeline.Service
+
+			defService = func() (*mocks.MockService, *unstructured.Unstructured) {
+				service := mocks.NewMockService(mockCtrl)
+				serviceContent := &unstructured.Unstructured{}
+				service.EXPECT().Resource().Return(serviceContent)
+				services = append(services, service)
+				return service, serviceContent
+			}
+			err = errors.New("e")
+		)
+
+		BeforeEach(func() {
+			services = []pipeline.Service{}
+			ctx.EXPECT().Services().DoAndReturn(func() ([]pipeline.Service, error) { return services, nil })
+		})
+
+		It("should collect from secret referred in services", func() {
+			secretName1 := "foo"
+			secretName2 := "bar"
+			ns1 := "ns1"
+			ns2 := "ns2"
+			svc1, content1 := defService()
+			content1.Object = map[string]interface{}{
+				"status": map[string]interface{}{
+					"binding": map[string]interface{}{
+						"name": secretName1,
+					},
+				},
+			}
+			content1.SetNamespace(ns1)
+			content1.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Secret"})
+			content1.SetName(secretName1)
+
+			svc2, content2 := defService()
+			content2.Object = map[string]interface{}{
+				"status": map[string]interface{}{
+					"binding": map[string]interface{}{
+						"name": secretName2,
+					},
+				},
+			}
+			content2.SetNamespace(ns2)
+			content2.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Secret"})
+			content2.SetName(secretName2)
+
+			ctx.EXPECT().ReadSecret(ns1, secretName1).Return(content1, nil)
+			ctx.EXPECT().ReadSecret(ns2, secretName2).Return(content2, nil)
+
+			ctx.EXPECT().AddBindings(&pipeline.SecretBackedBindings{Service: svc1, Secret: content1})
+			ctx.EXPECT().AddBindings(&pipeline.SecretBackedBindings{Service: svc2, Secret: content2})
+
+			collect.DirectSecretReference(ctx)
+		})
+
+		It("should retry processing if reading secret fails", func() {
+			_, content := defService()
+			secretName1 := "foo"
+			ns1 := "ns1"
+			content.SetName(secretName1)
+			content.SetNamespace(ns1)
+			content.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Secret"})
+			content.SetName(secretName1)
+
+			ctx.EXPECT().ReadSecret(ns1, secretName1).Return(nil, err)
+
+			ctx.EXPECT().RetryProcessing(err)
+			ctx.EXPECT().SetCondition(v1alpha1.Conditions().NotCollectionReady().Reason(collect.ErrorReadingSecret).Msg(err.Error()).Build())
+			collect.DirectSecretReference(ctx)
+		})
+
+		It("continue processing if secret has annotations", func() {
```

Comment:
stages are arranged in the pipeline, so not sure what do you mean by "continue processing"?  The case is here that a secret will not be considered as direct secret references if it has a service.binding annotation attached to it. Could you please rephrase the test name to reflect that?

Reply:
How about something like this?
- `skip processing if the secret has annotations`
- `ensure secret with annotations are ignored`
Please suggest.

Reply:
`Ignore secret having a secret binding annotation`

Reply:
Done. 387c9cc3e38705018063de2dea18001a762885d7